### PR TITLE
Firestore tests

### PR DESCRIPTION
**Pull request by quiro91**
## Problem

The previous PR about Firestore with speakers and event details had no test

## Solution

We added tests for the mappers

### Test(s) added 

All the mappers

### Paired with 

Nobody

### CODE REVIEW

File: app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+class FirestoreEventMapperTest {
+
+    @Mock
+    lateinit var checksum: Checksum
+
+    lateinit var timeZone: DateTimeZone
+
+    @Before
+    fun before() {
+        MockitoAnnotations.initMocks(this)
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt]: We should use the `@Rule` instead! See https://static.javadoc.io/org.mockito/mockito-core/2.13.0/org/mockito/junit/MockitoRule.html (or other tests in the project)

File: app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    @Test
+    fun `event track should match after mapping`() {
+        val firestoreTrack = fixtureOfFirestoreTrack()
+        val firestoreEvent = fixtureOfFirestoreEvent(track = firestoreTrack)
+        val event = firestoreEvent.toEvent(checksum, timeZone)
+        Assert.assertEquals(firestoreTrack.toTrack(), event.track.get())
+    }
+
+    @Test
+    fun `event track should be absent when is null`() {
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt]: Nit, here and elsewhere: could you change `when is null` with `when mapping null`?

File: app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    @Before
+    fun before() {
+        MockitoAnnotations.initMocks(this)
+        `when`(checksum.getChecksumOf(FAKE_STRING)).thenReturn(CHECKSUM_RESULT)
+        timeZone = DateTimeZone.UTC
+    }
+
+    @Test
+    fun `event id should match after mapping`() {
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt]: Nit, here and elsewhere, could you change `X after mapping` with `when mapped`?

File: app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    @Test
+    fun `event experience level should be absent when is null`() {
+        val firestoreEvent = fixtureOfFirestoreEvent(experienceLevel = null)
+        val event = firestoreEvent.toEvent(checksum, timeZone)
+        Assert.assertFalse(event.experienceLevel.isPresent)
+    }
+
+    @Test
+    fun `event timezone should match the parameter`() {
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreEventMapperTest.kt]: Nit: consistency would dictate a `when mapped` at the end

File: app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+import net.squanchy.service.firestore.model.conferenceinfo.FirestoreVenue
+import net.squanchy.service.firestore.model.schedule.FirestoreDay
+import net.squanchy.service.firestore.model.schedule.FirestoreEvent
+import net.squanchy.service.firestore.model.schedule.FirestorePlace
+import net.squanchy.service.firestore.model.schedule.FirestoreSchedulePage
+import net.squanchy.service.firestore.model.schedule.FirestoreSpeaker
+import net.squanchy.service.firestore.model.schedule.FirestoreTrack
+import java.util.Date
+
+fun fixtureOfFirestoreTrack(
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt]: I love these fixtures 💯 Could we name these `aFirebaseTrack`, `aFirestoreSpeaker` (and so on) for terseness?

File: app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+fun fixtureOfFirestoreSpeaker(
+    id: String = "speakerId",
+    name: String = "Boaty McBoatface",
+    bio: String = "Hello, I'm a fake",
+    companyName: String? = "Squanchy",
+    companyUrl: String? = "www.squanchy.net",
+    personalUrl: String? = "www.squanchy.net",
+    photoUrl: String? = "@DonuldTrump",
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt:39]: Donuld? LOL <3

File: app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    name: String = "Turin",
+    floor: String? = "This floor"
+) = FirestorePlace().apply {
+    this.id = id
+    this.name = name
+    this.floor = floor
+}
+
+fun fixtureOfFirestoreDay(
+    id: String = "today",
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt]: This is a bit confusing, can you use `dayId` as a value please?

File: app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    date: Date = Date()
+) = FirestoreDay().apply {
+    this.id = id
+    this.date = date
+}
+
+fun fixtureOfFirestoreVenue(
+    name: String = "VenueOfTheConference",
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt]: Nit: put spaces in the value so it doesn't look like we don't support spaces in there?

File: app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    this.latLon = latLon
+    this.description = description
+    this.mapUrl = mapUrl
+    this.timezone = timezone
+}
+
+fun fixtureOfFirestoreEvent(
+    id: String = "EventId",
+    title: String = "Getting started with fakes",
+    startTime: Date = Date(),
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt]: ⚠️ these should be set manually to some value (e.g., `startTime: Date = Date(12340000)` and `endTime: Date = Date(12345678)`) so that you have an ensured positive and non-zero duration for the event. If you don't you may hit edge cases in tests where you're not supposed to

File: app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+    floor: String? = "This floor"
+) = FirestorePlace().apply {
+    this.id = id
+    this.name = name
+    this.floor = floor
+}
+
+fun fixtureOfFirestoreDay(
+    id: String = "today",
+    date: Date = Date()
```
Comment (rock3r) [inline app/src/test/java/net/squanchy/service/firestore/FirestoreFixtures.kt]: ⚠️ You should use a fixed value! Leaving it like this may make the tests hit edge cases on some machines. Make it work with the defaults you specify in the `FirestoreEvent` down below (as in, they would be on the same day)

**Review by rock3r — CHANGES_REQUESTED** [review]
Great tests! I have concerns about the `Date()`s which may bring instability, and a few opinions on naming :D (of course I have opinions, but it's for consistency)

**Comment by quiro91** [comment]
@rock3r ready for review again

**Review by rock3r — APPROVED** [review]
Good for me now!
